agent/proxycfg/state.go

```diff
@@ -1518,3 +1541,35 @@ func (s *state) Changed(ns *structs.NodeService, token string) bool {
 		!reflect.DeepEqual(s.proxyCfg, proxyCfg) ||
 		s.token != token
 }
+
+// When endpoints come in for mesh gateways or external services we check to see if any instances have a hostname their address.
```

Comment:
nit: I still prefer to format method comments according to go conventions even for private methods. This comment is useful context but perhaps we could reword the first line to match conventional way of describing the intent of a method?

```suggestion
// hostnameEndpoints returns all CheckServiceNodes that have hostnames instead of IPs as the address.
//
```

Reply:
Sounds good to me. Fixed.